### ISSUE TITLE
Unable to Run in react strict mode / concurrent mode

### ISSUE THREAD

**Issue by michaelspeed** [prescottprue/react-redux-firebase#564]
**Do you want to request a *feature* or report a *bug*?**  
bug

With react 16.7 alpha 0, enabling concurrent mode gives me this error - 

`Warning: Unsafe lifecycle methods were found within a strict-mode tree:

componentWillMount: Please update the following components to use componentDidMount instead: FirebaseConnect(Connect(Core))

componentWillReceiveProps: Please update the following components to use static getDerivedStateFromProps instead: Connect(Core), FirebaseConnect(Connect(Core)), Provider

componentWillUpdate: Please update the following components to use componentDidUpdate instead: Connect(Core)`

**Comment by michaelspeed** [comment]
but still the last warning is of {Connect} from react-redux package. i am not sure they are also planning to move to componentDidMount()

**Comment by prescottprue** [comment]
That is correct, though I am assuming that they will.

**Comment by michaelspeed** [comment]
thanks going to try this now

**Comment by michaelspeed** [comment]
no exported member 'ReactReduxFirebaseProvider'
with type script? index.d.ts not updated?

**Comment by prescottprue** [comment]
@michaelspeed Yeah, the context's themselves were exported, so you can do `ReactReduxFirebaseContext.Provider`, but I also am planning on a releasing a change with the providers exported. Thanks for reaching out.

**Comment by prescottprue** [comment]
@cvanem I made some changes and published v3.0.0-alpha.4. Can you give it a shot now and see if it works as expected?

**Comment by cvanem** [comment]
@prescottprue Thanks, fixed the typescript error.  Firebase is showing up in the redux store, but firebase.auth.isLoaded always equals false and I can't get authIsReady to fire.  It may be something configured incorrectly on my end.

This is how my reducer is configured, which I believe is correct: 

    firebase.initializeApp(firebaseConfig);
    firebase.auth();
...

    function buildRootReducer(allReducers, history) {
      return combineReducers<ApplicationState>(Object.assign({}, allReducers, { router: (history) }));
    }


Also, are we still still suppose to use withFirebase to access the context or is there a better way now?

**Comment by cvanem** [comment]
@prescottprue It looks like I needed to add the initializeAuth property.  This worked:

    const rrfProps = {
      firebase,
      config: rrfConfig,
      dispatch: store.dispatch,
      initializeAuth: true,  
    };

**Comment by prescottprue** [comment]
@cvanem You shouldn't have to pass the initializeAuth prop yourself - seems like it was a misspelling in the default prop, so I am going to release a fix for that.

As for getting from context - Yes, `withFirebase` and `firebaseConnect` are both still good ways to get the wrapped Firebase instance. It is also available as a render prop if you use `ReactReduxFirebaseContext.Consumer` like so:

```jsx
<ReactReduxFirebaseContext.Consumer>
   {firebase => (
   <SomeProp
      firebase={firebase}
      dispatch={firebase.dispatch}
    />
  )}
</ReactReduxFirebaseContext.Consumer>
```

**Comment by cvanem** [comment]
@prescottprue Ok, good to know.  I tested v3.0.0 alpha.5 and everything is working as expected (without manually setting initializeAuth).  Thanks.
